src/hooks/useImmediateState.ts

```diff
@@ -0,0 +1,32 @@
+import { useRef, useEffect, useState, useCallback } from 'react';
+
+type SetStateAction<S> = S | ((prevState: S) => S);
+type SetState<S> = (action: SetStateAction<S>) => void;
+
+export const useImmediateState = <S extends {}>(init: S): [S, SetState<S>] => {
+  const isMounted = useRef(false);
+  const initialState = useRef<S>({ ...init });
+  const [state, setState] = useState<S>(initialState.current);
+
+  const updateState: SetState<S> = useCallback((action: SetStateAction<S>) => {
+    // If the component is currently before its initial mount, update
```

Comment:
For this comment shouldn't it be !isMounted.current?

Reply:
Oh the comment goes for the entire block; I'll pull it out of useCallback to clarify

Reply:
Ah allright, sorry got confused

Reply:
No worries 👌 I'll add more comments